packages/react-link/src/components/Link/Link.types.ts

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import { ComponentProps } from '@fluentui/react-utils';
+import { ColorTokens, ColorTokenSet, FontTokens, RecursivePartial } from '@fluentui/theme';
+
+/**
+ * {@docCategory Link}
+ */
+export type LinkProps = ComponentProps &
+  React.AnchorHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | HTMLElement> &
+  Omit<React.ButtonHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | HTMLElement>, 'type'> & {
+    /**
+     * URL the link points to. If not provided, the link renders as a button (unless that behavior is
+     * overridden using `as`).
+     */
+    href?: string;
+
+    /**
+     * Where to open the linked URL. Common values are `_blank` (a new tab or window),
+     * `_self` (the current window/context), `_parent`, and `_top`.
+     * This prop is only applied if `href` is set.
+     */
+    target?: string;
+
+    /**
+     * Relationship to the linked URL (can be a space-separated list).
+     * Most common values are `noreferrer` and/or `noopener`.
+     * This prop is only applied if `href` is set.
+     */
+    rel?: string;
+
+    /**
+     * Click handler for the link.
+     */
+    onClick?: (event: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement | HTMLElement>) => void;
+
+    /**
+     * Whether the link is disabled.
+     * @defaultvalue false
+     */
+    disabled?: boolean;
+
+    /**
+     * When set, allows the link to be focusable even when it has been disabled. This is used in scenarios where it is
+     * important to keep a consistent tab order for screen reader and keyboard users.
+     * @defaultvalue false
+     */
+    disabledFocusable?: boolean;
+
+    /**
+     * Built-in HTML attribute with different behavior depending on how the link is rendered.
+     * If rendered as `<a>`, hints at the MIME type.
+     * If rendered as `<button>`, override the type of button (`button` is the default).
+     */
+    type?: string;
+
+    /**
+     * If true, changes styling when the link is being used alongside other text content.
+     */
+    inline?: boolean;
+
+    /** Style tokens */
+    tokens?: RecursivePartial<LinkTokens>;
+  };
+
+/**
+ * {@docCategory Link}
+ */
+export interface LinkState extends LinkProps {
+  linkRef?: React.RefObject<HTMLElement>;
```

Comment:
```suggestion
  ref: React.RefObject<HTMLElement>;
```

Oops, this should be just `ref`

Reply:
This is actually a problem in the code scaffolding that hopefully @bsunderhus will help use with 😄

Reply:
I've checked and it seems that the `ref` is not needed at all in here, so I've removed it for now.